### PR TITLE
Support for PyWavelets.

### PR DESCRIPTION
**Pull request by AGrigis**
* Add support to PyWavelets: all the transforms from pywt are now accessible from pysap.
* Add a function to filter available transforms: as we have many transforms available in pysap, add a simple tool to filter these transforms.

### CODE REVIEW

File: pysap/extensions/transform.py [hunk 1]
```diff
@@ -10,69 +10,376 @@
...
+    else:
+        func = pywt.Wavelet
+    for name in pywt.wavelist(family):
+        pywt_class_factory(func, name, destination_module_globals)
 
 
 class ISAPWaveletTransformBase(WaveletTransformBase):
     """ Define the structure that will be used to store the ISAP results.
     """
+    __family__ = "isap"
```
Comment (LElgueddari) [inline pysap/extensions/transform.py]: changing name to "isap-2d"?

File: pysap/extensions/transform.py [hunk 2]
```diff
@@ -10,69 +10,376 @@
...
+            else:
+                raise NameError("Please define a correct dimension for data.")
+        else:
+            if self.data_dim == 2:
+                self.trf = None
+            else:
+                raise NameError("For {0}D, only the bindings work for "
+                                "now.".format(self.data_dim))
+
+    def _analysis(self, data, **kwargs):
```
Comment (LElgueddari) [inline pysap/extensions/transform.py]: Should we add an "axes" parameter which could be helpful for multichannel reconstruction for example

File: pysap/test/test_binding.py [hunk 3]
```diff
@@ -33,8 +33,11 @@ def setUp(self):
             get_sample_data(dataset_name="mri-slice-nifti")]
         print("[info] Image loaded for test: {0}.".format(
             [i.data.shape for i in self.images]))
+        transforms_struct = pysap.wavelist(["isap", "isap-3d"])
+        transforms_names = (
+            transforms_struct["isap"] + transforms_struct["isap-3d"])
```
Comment (LElgueddari) [inline pysap/test/test_binding.py]: Same for isap-2d

File: pysap/extensions/transform.py (hunk 1 above)
Comment (AGrigis) [reply pysap/extensions/transform.py]: I changed the family name.

File: pysap/test/test_binding.py (hunk 3 above)
Comment (AGrigis) [reply pysap/test/test_binding.py]: I changed the family name.

File: pysap/extensions/transform.py [hunk 4]
```diff
@@ -10,69 +10,376 @@
...
-
-For 3D:
-
-- BiOrthogonalTransform3D
-- Wavelet3DTransformViaLiftingScheme
-- ATrou3D
+Available 2D transform from ISAP are:
 
+- to get the full list of builtin wavelets' names just use the pysap.wavelist
+  with 'isap' as the family argument.
```
[the remark this answers is not in the document]
Comment (AGrigis) [reply pysap/extensions/transform.py]: I changed the family name.

File: pysap/extensions/transform.py [hunk 5]
```diff
@@ -10,69 +10,376 @@
...
+    }
+
+    # Get the process instance associated to the function
+    destination_module_globals[class_name] = (
+        type(class_name, (PyWaveletTransformBase, ), class_parameters))
+
+
+destination_module_globals = globals()
+for family in pywt.families():
+    if family in ("gaus", "mexh", "morl", "cgau", "shan", "fbsp", "cmor"):
```
[the remark this answers is not in the document]
Comment (AGrigis) [reply pysap/extensions/transform.py]: I kept this part of the code (2 lines).
In the future if we really do not want to use continuous wavelet transforms, we will raise an exception.

File: pysap/extensions/transform.py [hunk 6]
```diff
@@ -10,69 +10,376 @@
...
 import numpy
+import pywt
+
+
+class PyWaveletTransformBase(WaveletTransformBase):
+    """ Define the structure that will be used to store the pywt results.
+    """
+    __family__ = "pywt"
+
+    def __init__(self, nb_scale, verbose=0, dim=2, is_decimated=True,
```
[the remark this answers is not in the document]
Comment (AGrigis) [reply pysap/extensions/transform.py]: I kept the exact same parameters for ISAP and PYWT -> dim parameter.
I  added an axes parameter that is specific to PYWT.

File: pysap/extensions/transform.py (hunk 2 above)
Comment (AGrigis) [reply pysap/extensions/transform.py]: I  added an axes parameter that is specific to PYWT.

**Pull request merged**